src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,187 @@
+        this.size = 0;
+        first = null;
+        last = null;
+    }
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            addFirst(value);
+        } else {
+            addLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (!(index >= 0 && index <= size)) {
+            throw new IndexOutOfBoundsException("Wrong index value");
+        }
+        if (size == 0) { // item is actually the very first in the list
+            addFirst(value);
+        } else if (index == size) { // item is adding at the last position, next by this.last
+            addLast(value);
+        } else {
+            MyLinkedList.Node<T> positionNode = getPositionByIndex(index);
+            addBefore(value, positionNode);
+        }
+    }
+
+    private Node<T> getPositionByIndex(int index) {
+        if (index <= size / 2) {
+            return lookFromFirst(index);
+        } else {
+            return lookFromLast(index);
+        }
+    }
+
+    private Node<T> lookFromFirst(int index) {
+        MyLinkedList.Node<T> currentNode = this.first;
+        for (int i = 0; i < size; i++) {
+            if (i == index) {
+                break;
+            }
+            currentNode = currentNode.next;
+        }
+        return currentNode;
+    }
+
+    private Node<T> lookFromLast(int index) {
+        MyLinkedList.Node<T> currentNode = this.last;
+        for (int i = size - 1; i >= 0; i--) {
+            if (i == index) {
+                break;
+            }
+            currentNode = currentNode.prev;
+        }
+        return currentNode;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item: list) {
+            add(item);
+        }
+    }
+
+    private void addFirst(T value) {
+        MyLinkedList.Node<T> newNode = new MyLinkedList.Node<>(null, value, null);
+        this.first = this.last = newNode;
+        ++size;
+    }
+
+    private void addLast(T value) {
+        MyLinkedList.Node<T> newNode = new MyLinkedList.Node<>(this.last, value, null);
+        this.last.next = newNode;
+        this.last = newNode;
+        ++size;
+    }
+
+    private void addBefore(T value, Node<T> node) {
+        if (node == this.first) {
+            this.first = new MyLinkedList.Node<>(null, value, node);
+        } else {
+            MyLinkedList.Node<T> newNode = new MyLinkedList.Node<>(node.prev, value, node);
+            newNode.prev.next = newNode;
+            node.prev = newNode;
+        }
+        ++size;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (!(index >= 0 && index < size)) {
```

Comment:
```suggestion
        if (!(index >= 0 && index < size)) {
```

Why we cant get node on 0 index? Could we get node on index=size position?

Reply:
Умова тут якраз і говорить про те, що ми не можемо отримати ноду з індексом, який дорівнює розміру, і можемо, якщо індекс дорівнює нулеві.
Звісно, заморочено, і згоден, потрібно винести в окремий метод.
Подібна перевірка індекса провадиться при додаванні ноди, але там ми якраз можемо додати ноду, коли індекс дорівнює розмірові.